src/components/data/PowerBIReport/index.tsx

```diff
@@ -150,7 +156,17 @@ const PowerBIReport: React.FC<PowerBIProps> = ({ reportId, filters }) => {
 
                 embeddedRef.current.off('loaded');
                 embeddedRef.current.off('error');
-                embeddedRef.current.on('loaded', () => setIsLoading(false));
+                embeddedRef.current.on('loaded', () => {
+                    // console.log("Loaded Time:",(new Date().getTime() - timeLoadStart.getTime())/1000," Seconds")
+
+                    telemetryLogger.trackMetric({
+                        name: `${useCurrentApp.name}-EmbedloadedTime`,
+                        average: new Date().getTime() - timeLoadStart.getTime(),
+                        sampleCount: 1,
+                    });
+
+                    setIsLoading(false);
+                });
```

Comment:
There is also a "rendered" listener here that could be used for tracking. 
it triggers when the report is done rendering or user has interacted with the report. 

I that case i could add something like this : 

```
embeddedRef.current.on('rendered', () =>{
   telemetryLogger.trackMetric({
     name:`${useCurrentApp.name}-EmbedRenderedTime`,
     average: new Date().getTime() - timeLoadStart.getTime(),
     sampleCount:1
 });
```